### ISSUE TITLE
Service stops working without any error or message

### ISSUE THREAD

**Issue by tinodj** [blockchain/service-my-wallet-v3#202]
Hi,

I had this service working on a server for more than 6months continuously without any problem. I don't remember what version was that one, but the newest at that point. However, few days ago I installed new service, version 0.23.3. Node and npm are the same (4.2.6 and 2.5.11). Now the service stops/quits every few hours without any message or whatsoever in the logs. Here is the command how I start the service: 

nohup blockchain-wallet-service start --port 3030  >>/var/log/blockchain-wallet-service.log 2>&1 &

Can you please help me to find out what the reason for this is and to fix it? 
Can you suggest how I can install older version, and which one do you recommend?

**Comment by jtormey** [comment]
Hmm, that's strange, not sure why that would be happening. You can reset back to an older version using npm: `npm install -g blockchain-wallet-service@X.Y.Z` (replacing X.Y.Z with the version you want).

To help me with looking into this issue, could you give me an idea of how much stress is put on your service instance? How many wallets at a time, most frequent requests, etc. would be great.

**Comment by 1HVhD25fCP1pMXAo1bdxZtb3L6qQR9MqGD** [comment]
Moilui0609@gmail.com mickatiff@hotmail.fr.com mickatiff57@gmail.com

1262780

**Comment by tinodj** [comment]
Hi,

Thanks. It works now for 48 hours without any issue/crash/restart. It would be great if you put some logging on crash, so we can catch this for later debugging. Exiting silently generally is not a good idea :) 

There isn't lot of pressure: It is 1 wallet, and maybe about a transaction per minute, maybe less...

**Comment by jtormey** [comment]
You're right, more logging is a good idea, have been meaning to add that! Seems like a fairly random issue, possible memory leak? Not sure how else to explain a sudden crash, I'll investigate.

**Comment by jtormey** [comment]
@tinodj, what version of the service have you been using that works consistently over longer periods of time? Would be useful for me to know so I can compare versions.

**Comment by tinodj** [comment]
hi as I said in my first post: "I don't remember what version was that one, but the newest at that point.". I would say 0.20 (checked the history of releases). But, as I said earlier, now even this version is working 30days without any problem.

**Comment by jtormey** [comment]
Thanks for the quick response @tinodj. Just to clarify, the version you're using now (which has been working for 30 days without issue) is 0.23.3? Or 0.20? Thanks.

**Comment by tinodj** [comment]
Clarification: 

1. one making problems and causing me to open this issue is 0.23.3. And this one is working now for 30days. However I remember that with every version I had a crash (in the first 24hours) after the first start, but it was working well after the second start. But with this version it was the third start that is working now for 30days. I know how strange this sounds, because I am programmer myself, but it is like it is. 

2. one that was working for 6+ months was (I guess) 0.20.

**Comment by jtormey** [comment]
That is very strange, thanks for clarifying! Will run some tests of my own.

**Comment by adenisdeveloper** [comment]
I'm going through the same difficulty, mine is worse than it ends the service in less than 3 hours now.
